src/main.rs

```diff
@@ -29,9 +56,9 @@ fn build_menu(application: &gtk::Application, window: &gtk::ApplicationWindow) {
 
     // Connect the about menu entry click event.
     let about = gio::SimpleAction::new("about", None);
-    let weak_window = window.downgrade();
+    let weak_application = application.downgrade();
```

Comment:
I just thought that we could greatly improve the `upgrade_weak` macro to make it look like the `clone` macro (taking arguments and calling `upgrade` on all given arguments).

Reply:
I have that on my TODO list, a combined macro that does clone or downgrade based on a marker in the signature and also allows renaming things (so you can pass `self` from outside the closure into the closure, where `self` would be illegal).

Reply:
That'd be amazing! We definitely need a macro crate in gtk-rs for such things.